pkg/nuclide/file-tree/lib/FileTreeController.js

```diff
@@ -737,16 +743,50 @@ class FileTreeController {
       onClose: () => this._closeDialog(),
       shouldSelectBasename: true,
     };
-    this._openDialog(<FileDialogComponent {...props} />);
+    this._openDialog(props);
+  }
+
+  openDuplicateDialog(): void {
+    var selection = this._getSelectedEntryAndDirectoryAndRoot();
+    if (!selection || selection.entry.isDirectory()) {
+      return;
+    }
+
+    var message = (
+      <div>
+        <div>Enter the new name for the file in the root:</div>
+        <div>{path.normalize(selection.root.getPath() + '/')}</div>
+      </div>
+    );
+
+    var {entry, root} = selection;
+
+    var props = {
+      rootDirectory: root,
+      initialEntry: entry,
+      message,
+      onConfirm: async (rootDirectory, relativeFilePath) => {
+        var file = rootDirectory.getFile(relativeFilePath);
+        await file.create();
```

Comment:
If it couldn't be created, maybe create one more modal with the explanation that it couldn't be created with just an "OK" button? Any ideas there? We don't want to fail silently.